tools/steam-shell.pike

```diff
@@ -458,25 +458,22 @@ int list(string what)
     write("Wrong usage\n");
     return 0;
   }
-  int flag=0;
   string toappend="";
   array(string) display = get_list(what);
-  string a="";
   if(sizeof(display)==0)
-    toappend = "There are no "+what+" in this room\n";
-  else
+    toappend = "There are no "+what+" in this room.\n";
+  else if (display[0]=="Invalid command")
+  {
+    write(display+"\n");
```

Comment:
An array is given out as an output. Since pike is an interpreted language if any index exceeding the size of it are tried to be accessed, AIOB error would be thrown.